harp/reader.py

```diff
@@ -44,13 +44,53 @@ def __init__(
         self.read = read
 
 
+class RegisterMap:
+    registers: dict[str, RegisterReader]
```

Comment:
I would maybe make both dict instances private and maybe call them something like `_name_map`, `_address_map`. The idea is that the `RegisterMap` class itself is the data structure, so you only access it usually through the public interface.